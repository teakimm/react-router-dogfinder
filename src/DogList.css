.DogList {
  display: flex;
  justify-content: center;
  gap: 2em;
}