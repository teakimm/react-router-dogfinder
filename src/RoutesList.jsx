import React from "react";
import { Navigate, Route, Routes } from "react-router-dom";


function RoutesList() {
    return (
        <Routes>
            <Route path="*" element={<Navigate to="/dogs" />} />
        </Routes>
    );
}

export default RoutesList;