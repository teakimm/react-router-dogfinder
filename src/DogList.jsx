import './App.css';
import { BrowserRouter, Link } from 'react-router-dom';
import RoutesList from './RoutesList';
import { useState } from 'react';
import { v4 as uuid } from "uuid";
import './DogList.css';


/** Renders cards for all dogs
 *
 * state:
 * none
 *
 * props:
 * -dogs: [{name, age, ...}]
 *
 * App -> RoutesList -> DogList
 */
function DogList({ dogs }) {
    const dogCards = dogs.map(dog =>
        <div key={dog.name} className="card" style={{ width: '18em' }}>
            <img className="card-img-top" src={`/public/${dog.src}.jpg`} alt={dog.src} />
            <div className="card-body">
                <h5 className="card-title">{dog.name}</h5>
                <Link className="btn btn-primary" to={`/dogs/${dog.name.toLowerCase()}`}>{`More about ${dog.name}`}</Link>
            </div>
        </div>
    );

    return (
        <div className="DogList">{dogCards}</div>
    );
};
export default DogList;