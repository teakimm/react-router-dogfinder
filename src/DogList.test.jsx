import { describe, it, expect, vi } from "vitest";

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import DogList from "./DogList";;
import { MemoryRouter } from "react-router-dom";
import RoutesList from "./RoutesList";

const DOGS = [
    {
        "name": "Whiskey",
        "age": 5,
        "src": "whiskey",
        "facts": [
            "Whiskey loves eating popcorn.",
            "Whiskey is a terrible guard dog.",
            "Whiskey wants to cuddle with you!"
        ]
    },
    {
        "name": "Duke",
        "age": 3,
        "src": "duke",
        "facts": [
            "Duke believes that ball is life.",
            "Duke likes snow.",
            "Duke enjoys pawing other dogs."
        ]
    },
    {
        "name": "Perry",
        "age": 4,
        "src": "perry",
        "facts": [
            "Perry loves all humans.",
            "Perry demolishes all snacks.",
            "Perry hates the rain."
        ]
    }
];

describe("DogList component", function () {
    it("renders without crashing", function () {
        const { container } = render(
            <MemoryRouter initialEntries={["/dogs"]}>
                <RoutesList dogs={DOGS} />
            </MemoryRouter>
        );


    });

    it("matches snapshot", function () {
        const { container, debug } = render(
            <MemoryRouter initialEntries={["/dogs"]}>
                <RoutesList dogs={DOGS} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });

    it("renders correct dog information", function () {
        const result = render(
            <MemoryRouter initialEntries={["/dogs"]}>
                <RoutesList dogs={DOGS} />
            </MemoryRouter>
        );

        expect(result.queryByText("More about Perry")).toBeInTheDocument();
        expect(result.queryByText("More about Duke")).toBeInTheDocument();
        expect(result.queryByText("More about Whiskey")).toBeInTheDocument();

        expect(
            result.container.querySelectorAll(".card").length
        ).toEqual(3);
    });

    it("link to specific dog works", function () {
        const result = render(
            <MemoryRouter initialEntries={["/dogs"]}>
                <RoutesList dogs={DOGS} />
            </MemoryRouter>
        );

        const dukeLink = result.container.querySelector(".Duke");
        fireEvent.click(dukeLink);


        expect(result.queryByText("Duke believes that ball is life.")).toBeInTheDocument();
        expect(result.queryByText("Duke likes snow.")).toBeInTheDocument();
        expect(result.queryByText("Duke enjoys pawing other dogs.")).toBeInTheDocument();
    });
});
