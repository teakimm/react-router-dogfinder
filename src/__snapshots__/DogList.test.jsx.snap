// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DogList component > matches snapshot 1`] = `
<div>
  <div
    class="DogList"
  >
    <div
      class="card"
      style="width: 18em;"
    >
      <img
        alt="whiskey"
        class="card-img-top"
        src="/public/whiskey.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Whiskey
        </h5>
        <a
          class="btn btn-primary Whiskey"
          href="/dogs/whiskey"
        >
          More about Whiskey
        </a>
      </div>
    </div>
    <div
      class="card"
      style="width: 18em;"
    >
      <img
        alt="duke"
        class="card-img-top"
        src="/public/duke.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Duke
        </h5>
        <a
          class="btn btn-primary Duke"
          href="/dogs/duke"
        >
          More about Duke
        </a>
      </div>
    </div>
    <div
      class="card"
      style="width: 18em;"
    >
      <img
        alt="perry"
        class="card-img-top"
        src="/public/perry.jpg"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Perry
        </h5>
        <a
          class="btn btn-primary Perry"
          href="/dogs/perry"
        >
          More about Perry
        </a>
      </div>
    </div>
  </div>
</div>
`;
